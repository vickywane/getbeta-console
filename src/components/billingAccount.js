import React, { useState } from 'react'
import styled from 'styled-components'
import media from 'styled-media-query'
import { inject, observer } from 'mobx-react'
import { Stack, Select } from '@chakra-ui/core'
import * as Yup from 'yup'
import { Spinner } from 'react-bootstrap'

import useWindowWidth from '../utils/hook_style'
import {
  Text,
  Title,
  center,
  Alert,
  Button,
  Body,
  MdTitle,
  CreateCourseInputField as InputContainer
} from '../styles/style'

const Card = styled.div`
  width: auto;
  margin: 1rem 0;
  span {
    padding: 1rem 1rem;
  }
`

const InputBody = styled.div`
  margin: 1rem 1rem;
  label {
    font-size: 0.85rem;
  }
  input {
    display: flex;
    flex: 1;
    width: ${props => props.width};
    height: 50px;
    border: 1px solid #c0c0c0;
    color: black;
    padding: 0.5rem 1rem;
    border-radius: 3px;
    font-size: 0.9rem;
  }
  select {
    border: 1px solid #c0c0c0;
  }
  ${media.lessThan('large')`
  label {
   font-size : 1rem;
  }
  input {
    display: flex;
    flex: 1;
    width: ${props => props.width};
    height: 40px;
    border: 1px solid ${props => (props.borderColor ? props.borderColor : '#fff')};
    color: blue;
    padding: 0.5rem 1rem;
    border-radius: 3px;
    font-size: 0.9rem;
  }
  ${media.lessThan('medium')`
    margin: 1rem 1rem;
    label {
      font-size : .85rem;
    }
    input {
      font-size : .9rem;
      background: #fbfbfb;
      &:focus {
        box-shadow: 0 0 1.5px 1.5px #0072ce;
      }
    }
  `};
  ${media.lessThan('small')`
  margin: .5rem 1rem;
  label {
    font-size : .8rem;
  }
  input {
    font-size : .8rem;
  }
`};
  `};
`

const FlexInputs = styled.div`
  display: grid;
  grid-template-columns: 50% 50%;
  place-items: center;
  ${media.lessThan('medium')`
    width : 100%;
    display : flex;
    flex-direction : column;
    align-items : center;
  `}
`

const SmallCard = styled.div`
  ${media.lessThan('medium')`
    width : 95%;
  `};
  ${media.lessThan('small')`
  width : 97%;
`};
`

const billingSchema = Yup.object().shape({
  bankName: Yup.string(),
  cardHolderName: Yup.string(),
  accountNumber: Yup.string()
})

const AlertComponent = ({ show, error }) =>
  show && (
    <Alert style={{ ...center, borderRadius: '5px 5px 0 0', background: error && 'red' }}>
      <Text> Your billing account has been added. </Text>
    </Alert>
  )

const BillingAccountModal = props => {
  const [bank, setBank] = useState('Guaranty Trust Bank') //default
  const [accountName, setAccountName] = useState('')
  const [accountNumber, setAccountNumber] = useState('')
  const [cvv, setCvv] = useState('')
  const [expiry, setExpiry] = useState('')

  const Width = useWindowWidth()
  const {
    subscribeToPlan,
    accountIsUpgraded,
    isLoading,
    errorCreatingBillingAccount,
    createBillingAccount,
    accountIsBillingAccountModald
  } = props.UserStore

  const handleCheckout = () => {
    const isValidAccount = billingSchema.isValid({
      bankName: bank,
      cardHolderName: '',
      accountNumber: ''
    })

    // isValidAccount
    //   .then(() => {
    //   })
    //   .catch(e => console.log(e))

    createBillingAccount(accountName, bank, accountNumber)
  }

  return (
    <div>
      {Width >= 1000 ? (
        <div style={{ ...center }}>
          <div>
            <AlertComponent show={accountIsUpgraded} error={errorCreatingBillingAccount} />
            <Text align="center">
              Billing accounts are only setup once and are used to purchase items on GetBeta{' '}
            </Text>
            <form onSubmit={() => handleCheckout()}>
              <InputBody style={{ color: 'black' }}>
                <label style={{ color: '#fff' }}>BANK NAME</label>
                <Select onChange={e => setBank(e.target.value)}>
                  <option value="Guaranty Trust Bank"> Guaranty Trust Bank </option>
                  <option value="Stanbic Bank"> Stanbic Bank </option>
                  <option value="First Bank Nigeria"> First Bank Nigeria </option>
                </Select>
              </InputBody>

              <InputBody width="100%">
                <label> ACCOUNT NUMBER </label>
                <input
                  value={accountNumber}
                  onChange={e => setAccountNumber(e.target.value)}
                  placeholder="XXXX-XXXX-XXXX-XXXX"
                />
              </InputBody>

              <InputBody width="100%">
                <label> CARDHOLDER NAME</label>

                <input
                  value={accountName}
                  onChange={e => setAccountName(e.target.value)}
                  placeholder="Cardholder Name"
                />
              </InputBody>

              <FlexInputs>
                <InputBody width="100%">
                  <label> EXPIRY DATE </label>
                  <input
                    onChange={e => setExpiry(e.target.value)}
                    value={expiry}
                    placeholder="01-01"
                  />
                </InputBody>

                <InputBody width="100%">
                  <label> CVV </label>
                  <input onChange={e => setCvv(e.target.value)} value={cvv} placeholder="XX-XX" />
                </InputBody>
              </FlexInputs>
            </form>
            <div style={{ display: 'flex', justifyContent: 'center' }}>
              <Button
                onClick={() => handleCheckout()}
                style={{
                  width: '95%'
                }}
              >
                {isLoading ? 'Adding Details' : 'Add Details'}

                {isLoading && (
                  <div style={{ paddingLeft: '.7rem' }}>
                    <Spinner size="sm" animation="border" role="status" />
                  </div>
                )}
              </Button>
            </div>
            <br />
          </div>
        </div>
      ) : (
        <div style={{ ...center }}>
          <SmallCard>
            <AlertComponent show={accountIsUpgraded} error={errorCreatingBillingAccount} />
            <span>
              <form onSubmit={() => handleCheckout()}>
                <InputContainer>
                  <label>Bank Name</label>
                  <Select onChange={e => setBank(e.target.value)}>
                    <option value="Guaranty Trust Bank"> Guaranty Trust Bank </option>
                    <option value="Stanbic Bank"> Stanbic Bank </option>
                    <option value="Gtb"> First Bank Nigeria </option>
                  </Select>
                </InputContainer>

                <InputContainer borderColor="#c0c0c0" width="100%">
                  <label> Account Number </label>
                  <input onChange={e => {}} placeholder="Account Number" />
                </InputContainer>

                <InputContainer borderColor="#c0c0c0" width="100%">
                  <label> Cardholder Name </label>
                  <input placeholder="Cardholder Name" />
                </InputContainer>

                <div style={{ margin: '1rem 0', display: 'flex', justifyContent: 'center' }}>
                  <Button
                    onClick={e => {
                      e.preventDefault()
                      handleCheckout()
                    }}
                    style={{ width: '100%' }}
                  >
                    Add Details
                  </Button>
                </div>
              </form>
            </span>
          </SmallCard>
        </div>
      )}
    </div>
  )
}

export default inject('UserStore')(observer(BillingAccountModal))
