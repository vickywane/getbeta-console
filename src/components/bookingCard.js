import React, { useState } from 'react'
import styled from 'styled-components'
import { navigate } from '@reach/router'
import media from 'styled-media-query'
import { Box, Badge } from '@chakra-ui/core'

import { Text, Title } from '../styles/style'

const UserImage = styled.img`
  height: auto;
  width: 90px;
  cursor: pointer;
  object-fit: cover;
  border-radius: 3px 0 0 3px;
  ${media.lessThan('large')`
  height: auto;
  width: 100px;
  `};
  ${media.lessThan('medium')`
    height: 150px;
    width: 150px;
    border-radius : 50%;
  `};
  ${media.lessThan('small')`
    height: 120px;
    width: 125px;
      border-radius : 5%;
  `};
`

const StyledCard = styled.div`
  width: 30rem;
  height: auto;
  border-radius: 7px;
  background: #fff;
  box-shadow: 0 1px 2px #c0c0c0;
  span {
    display: grid;
    grid-template-columns: 8rem auto;
    grid-gap: 0 0.5rem;
  }
  ${media.lessThan('large')`
      width : 27rem;
  `};
  ${media.lessThan('medium')`
    width: 29rem;
    span {
      padding : 1rem .6rem;
      display: flex;
      flex-direction : column;
      align-items : center;
    }
  `};
  ${media.lessThan('small')`
    width: 20rem;
  `};
`

const BookingCard = props => {
  const { fullname, _id, rating, email, price, setBookingModal, setDetails } = props

  return (
    <StyledCard>
      <span>
        <div style={{ display: 'flex', justifyContent: 'center' }}>
          <UserImage alt="User" src={require('../assets/images/img.jpg')} />
        </div>

        <div style={{ padding: '.5rem 0' }}>
          <div
            style={{
              display: 'flex',
              justifyContent: 'space-between',
              padding: '0 .7rem'
            }}
          >
            <Title
              onClick={() => {
                navigate(`/u/${fullname.trim()}`, {
                  state: { id: _id }
                })
              }}
              style={{ fontWeight: 500, cursor: 'pointer' }}
              color="#0072ce"
              small
              align="center"
            >
              {fullname}
            </Title>
          </div>

          <Text style={{ padding: '0', margin: '0' }} small align="center">
            Lorem ipsum dolor, sit amet consectetur adipisicing elit.{' '}
          </Text>

          <br />

          {/* <div style={{ display: 'flex', justifyContent: 'space-between' }}>
                  <Text small> Slots : 5 </Text>

                  <Text small style={{ color: '#0072ce' }}>
                    ${price} / mins{' '}
                  </Text>
                </div> */}

          <div style={{ display: 'flex', marginTop: '10px', justifyContent: 'space-between' }}>
            <Box d="flex" alignItems="baseline">
              <Badge rounded="full" px="5" fontSize={11} variantColor="teal">
                5 Open Slots
              </Badge>
            </Box>

            <Text
              style={{ paddingRight: '10px' }}
              align="center"
              onClick={() => {
                setBookingModal(true)
                setDetails({
                  fullname: fullname,
                  price: price,
                  email: email,
                  rating: rating
                })
              }}
            >
              Create Booking
            </Text>
          </div>
        </div>
      </span>
    </StyledCard>
  )
}

export default BookingCard
