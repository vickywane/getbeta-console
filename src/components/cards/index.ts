export { default as ProfileCard } from './profileCard'
export { default as StreamCard } from './streamCard'
