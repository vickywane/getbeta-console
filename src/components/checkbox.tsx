import React, { useState } from "react"
import styled from "styled-components"

const Check = styled.input`
	opacity: 0;
`

const CheckBody = styled.div`
		border-radius : 5px;
	svg {
		transition: all 400ms;
		width: 2rem;
		stroke: blue;
		stroke-width: 3rem;
		fill: white;
		border-radius : 5px;
		padding: 0.2rem 0.2rem;
	transition: all 300ms;

		rect {
			fill: ${props => (props.active ? "#f2f2f2" : "#fff")};
		}
		polyline {
			transition: all 300ms;
			opacity: ${props => (props.active ? 1 : 0)};
		}
	}
`

interface CustomProps {
	name: string
	handleClick: any
	backgroundColor?: string
}

const CheckBox = props => {
	const [Clicked, Click] = useState<boolean>(false)

	const { name, handleClick, backgroundColor } = props

	function handleChange(e, name) {
		e.preventDefault()
		let value: string

		// Clicked is false by default hence an inverse logic.
		if (Clicked) {
			value = "off"
		} else {
			value = "on"
		}

		Click(!Clicked)
		handleClick(value, name)
	}

	return (
		<div onClick={e => handleChange(e, name)}>
			<CheckBody
				backgroundColor={backgroundColor === undefined ? null : backgroundColor}
				active={Clicked}
			>
				<svg viewBox="0 0 475 446">
					<defs></defs>
					<title>Check box</title>
					<g id="Layer_2" data-name="Layer 2">
						<g id="Layer_1-2" data-name="Layer 1">
							<rect x="6" y="6" width="463" height="434" rx="6.47" />
							<path
								style={{ fill: "#2f00a8" }}
								d="M462.53,12a.47.47,0,0,1,.47.47V433.53a.47.47,0,0,1-.47.47H12.47a.47.47,0,0,1-.47-.47V12.47a.47.47,0,0,1,.47-.47H462.53m0-12H12.47A12.47,12.47,0,0,0,0,12.47V433.53A12.47,12.47,0,0,0,12.47,446H462.53A12.47,12.47,0,0,0,475,433.53V12.47A12.47,12.47,0,0,0,462.53,0Z"
							/>
							<polyline
								style={{
									padding: "1rem",
									stroke: "#2f00a8",
									strokeLinecap: "square",
									strokeLinejoin: "bevel",
									strokeWidth: "15px",
								}}
								points="105.53 216.26 213.18 307.99 392.57 91.97"
							/>
						</g>
					</g>
				</svg>
			</CheckBody>
		</div>
	)
}

export default CheckBox
