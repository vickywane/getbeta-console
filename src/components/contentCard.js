import React from 'react'
import { Box, Image, Badge } from '@chakra-ui/core'
import styled from 'styled-components'
import moment from 'moment'
import { navigate } from '@reach/router'

const property = {
  imageUrl:
    'https://res.cloudinary.com/dkfptto8m/image/upload/v1603528071/freelance/placeholer.png',
  imageAlt: 'placeholder for a content file',
  beds: 3,
  baths: 2,
  formattedPrice: '$1,900.00',
  reviewCount: 0,
  rating: 4
}

const StarIcon = styled.div`
  color: ${props => props.color};
`

function ContentCard(props) {
  const { id, createdAt, descrp, price, type, contentfiles, vendorId, subscribers, title } = props

  return (
    <div>
      <Box
        style={{ border: '1px solid #c0c0c0' }}
        maxW="sm"
        w={'80%'}
        borderWidth="2px"
        rounded="lg"
        overflow="hidden"
      >
        <Image
          height={150}
          objectFit="cover"
          size={'100%'}
          src={property.imageUrl}
          alt={property.imageAlt}
        />

        <Box p="6">
          <Box d="flex" alignItems="baseline">
            <Badge rounded="full" px="2" variantColor="teal">
              {type}
            </Badge>
          </Box>

          <Box
            onClick={() =>
              navigate('/edit-content', {
                state: {
                  contentId: id
                }
              })
            }
            style={{ cursor: 'pointer' }}
            mt="1"
            fontWeight="semibold"
            as="h6"
            lineHeight="tight"
            isTruncated
          >
            {title}
          </Box>

          <Box
            color="gray.500"
            fontWeight="semibold"
            letterSpacing="wide"
            fontSize="xs"
            textTransform="uppercase"
            ml="2"
          >
            {/* THROWS ERROR FOR USERS WHO USED THE APP BEFORE CONTENTFILES WHERE ADDED */}
            {/* {contentfiles && contentfiles.length} Files &bull; {subscribers.length} Subscribers */}
          </Box>

          <div style={{ display: 'flex', justifyContent: 'space-between' }}>
            <Box>
              $ {price}
              <Box as="span" color="gray.600" fontSize="sm">
                {moment(createdAt).format(' DD, MMM')}
              </Box>
            </Box>

            <Box d="flex" mt="2" alignItems="center">
              {Array(5)
                .fill('')
                .map((_, i) => (
                  <StarIcon key={i} color={i < property.rating ? 'teal.500' : 'gray.300'} />
                ))}
              <Box as="span" ml="2" color="gray.600" fontSize="sm">
                {property.reviewCount} reviews
              </Box>
            </Box>
          </div>
        </Box>
      </Box>
    </div>
  )
}

export default ContentCard
