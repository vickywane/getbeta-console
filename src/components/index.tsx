export { default as Header } from "./header/header"
export { default as Footer } from "./footer"
export { default as Loader } from "./loaders"
export { default as Detail } from "./detail"
export { default as ResolutionError } from "./ResolutionError"
export { default as Panes } from "./panes/pane"
export { default as SettingsPane } from "./panes/settings"
export { default as Switch } from "./switch"
export { default as Checkbox } from './checkbox'
export { default as Tip } from './tip'
