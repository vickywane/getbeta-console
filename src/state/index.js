// export {  store as UserStore } from './user.store'
export { default as CourseStore } from './courses.store'
export { default as VendorStore } from './vendor.store'
export { default as ContentStore } from './content.store'
