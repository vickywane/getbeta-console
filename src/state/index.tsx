export { default as ConsoleStore } from "./console.store";
export { default as MediaStore } from "./media.store";
export { default as AuthStore } from "./auth.store";
