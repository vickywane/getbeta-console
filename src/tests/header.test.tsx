import React from "react"
import { Header } from "../components/"
import { render } from "@testing-library/react"
import { AuthStore, ConsoleStore } from "../state"
import Enzyme, { shallow } from "enzyme"
import Adapter from "enzyme-adapter-react-16"

Enzyme.configure({ adapter: new Adapter() })

test("Header contains specific links", () => {
  const component = shallow(
    <Header
      AuthStore={AuthStore}
      ConsoleStore={ConsoleStore}
      page={"non-app"}
    />
  )

  let tree = component

  expect(tree).toMatchSnapshot()

  const { getByText } = render(
    <Header
      AuthStore={AuthStore}
      ConsoleStore={ConsoleStore}
      page={"non-app"}
    />
  )

  const linkElement = getByText(/Oasis/i)
  expect(linkElement).toBeInTheDocument()
})
