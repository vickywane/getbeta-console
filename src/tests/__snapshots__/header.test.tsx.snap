// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header contains specific links 1`] = `ShallowWrapper {}`;
